lib/Semantics/expression.cpp

```diff
@@ -1364,6 +1364,11 @@ MaybeExpr ExpressionAnalyzer::Analyze(
         Say(source,
             "Keyword '%s=' does not name a component of derived type '%s'"_err_en_US,
             source, typeName);
+      } else if (const auto *currScope{
+                     context_.globalScope().FindScope(source)}) {
+        if (auto msg{CheckAccessibleComponent(*currScope, *symbol)}) {
+          Say(source, *msg);
+        }
```

Comment:
I think this check would also work if the code was trying to initialized private components without keywords if you moved this check right after: https://github.com/flang-compiler/f18/blob/2d853b4b9e2ce4b41daa00efb56834be2b6f4276/lib/Semantics/expression.cpp#L1409

Reply:
Good idea, I've done that.